src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -830,6 +857,81 @@ evalApp γ ctx e0 es et
       , f == "Language.Haskell.Liquid.ProofCombinators.pleUnfold"
       = arg
       | otherwise = e
+-- **ETA**: PLE before the patch to perform eta expansion was only able to unfold
+-- fully applied functions, by eta equivalence we mean the property that for all
+-- functions f the terms f and (\x -> f x) have the same semantics, we leverage 
+-- this fact to fully apply functions with extra function arguments so to trigger
+-- the standard PLE espansion, see the following example:
+-- We have a function f:
+-- define f (x : int) : int = {(x)}
+-- And we need to prove some constraint of this shape
+-- { f = something }
+-- The original PLE cannot do anything on f as it's not fully applied, instead
+-- if we perform eta expansion on f we obtain the following constraint
+-- { \y:Int -> f y = something }
+-- And now PLE can perform the unfolding of f as it's now fully saturated
+-- { \y:Int -> y = something }
+-- Clearly we need the higerorder flag active as we are generating lambda in
+-- the constraints.
+evalApp γ _ctx e0 es _et
+  | EVar f <- dropECst e0
+  , Just eq <- Map.lookup f (knAms γ)
+  = do
+    isHOAllowed <- gets hoFlag
+    if isHOAllowed then do
+      let expectedArgs = eqArgs eq
+      let nProvidedArgs = length es
+      let nArgsMissing = length expectedArgs - nProvidedArgs
+
+      -- Fresh names for the eta expansion
+      etaNames <- makeFreshEtaNames nArgsMissing
+      -- For some reason the application is expected to be already 
+      -- monomorphized, so we can't extract the sorts directly form
+      -- the equation
+      let etaArgsMono = drop nProvidedArgs <$> extractMonoSign e0
+      case etaArgsMono of
+          -- This case is pretty sus as we shouldnt generate funcitons that do not
+          -- have a type annotation
```

Comment:
Sus meant suspicious, it' because at the beginning I tough that I was removing some `ECst` from some variables in some way but actually later I found out that actually they are already generated without the ECst in the previous passes

Reply:
I think that I remove that comment since its not my fault